R/html-tweak.R

```diff
@@ -571,6 +570,61 @@ tweak_404 <- function(html, pkg = pkg) {
   TRUE
 }
 
+# Add syntax highlighting to reference sections
+tweak_reference_topic_html <- function(html) {
+  switch_block_content <- function(block, highlighted_html) {
+    xml2::xml_text(block) <- ""
+      purrr::walk(highlighted_html, function(x) {xml2::xml_add_child(block, x) })
+  }
+
+  # blocks with language information = r
+  r_blocks <- xml2::xml_find_all(html, "//div[contains(@class, 'sourceCode r')]/pre/code")
+  # now add blocks with no language information
+  nolang_r_blocks <- xml2::xml_find_all(html, "//div[contains(@class, 'ref-section')]/pre/code")
+
+  highlight_r_block <- function(block) {
+    out <- downlit::highlight(
+      xml2::xml_text(block),
+      classes = downlit::classes_pandoc()
+    )
+    if (!is.na(out)) {
+      # Replace the original node contents with downlit output
+      highlighted_html <- xml2::xml_contents(
+        xml2::xml_find_first(xml2::read_html(out), "body")
+      )
+      switch_block_content(block, highlighted_html)
+    }
+  }
+
+  purrr::walk(r_blocks, highlight_r_block)
+  purrr::walk(nolang_r_blocks, highlight_r_block)
```

Comment:
Maybe, for easier tweaking later e.g. if the "adding copy-paste button" happens afterwards, the tweaking here should wrap these blocks into a div of class "sourceCode".

Reply:
Or maybe roxygen2 should do that.